### PR TITLE
sysconfig/cloudinit: Check if disabled via kernel commandline

### PR DESCRIPTION
**Pull request by holmanb**
Currently snapd manually checks for the presence of `/etc/cloud/cloud-init.disabled` to avoid running `cloud-init status`. This is done to avoid having to run `cloud-init status` to check current status. Similarly, snapd can check the kernel commandline for `cloud-init=disabled` which also [disables cloud-init permanently](https://cloudinit.readthedocs.io/en/latest/howto/disable_cloud_init.html#method-2-kernel-commandline).

A unit test is included.

### CODE REVIEW

File: sysconfig/cloudinit.go
Comment (bboozzoo) [inline sysconfig/cloudinit.go:753]: I think I'd log something if kernel command line cannot be parsed (it's unlikely we'd even reach this far in such case, but who knows):
```suggestion
	cmdline, err := kcmdline.KeyValues("cloud-init")
	if err != nil {
		logger.Noticef("cannot obtain cloud-init from kernel command line: %v", err)
	} else if err == nil && cmdline["cloud-init"] == "disabled" {
		return CloudInitDisabledPermanently, nil
	}
```

Comment (holmanb) [reply sysconfig/cloudinit.go:753]: Sure, seems reasonable to me.

**Comment by holmanb** [comment]
@bboozzoo Thanks for the review. I just rebased on the tip of master and pushed the requested changes.

Ready for re-review.

File: sysconfig/cloudinit.go
Comment (bboozzoo) [inline sysconfig/cloudinit.go:751]: something is off with formatting here, please run `go fmt`

File: sysconfig/cloudinit.go
Comment (bboozzoo) [inline sysconfig/cloudinit.go:753]: sorry, should have caught this earlier:
```suggestion
	} else if cmdline["cloud-init"] == "disabled" {
```

**Review by bboozzoo — APPROVED** [review]
LGTM with a tweak

File: sysconfig/cloudinit.go
Comment (ernestl) [inline sysconfig/cloudinit.go:752]: Consider making this a warning `logger.Noticef("WARNING: cannot ...`

**Comment by holmanb** [comment]
It looks like this is waiting for a review from @alfonsosanchezbeato. I believe that I've addressed all requested changes. Please let me know anything else needs to change.

**Comment by ernestl** [comment]
@Meulengracht, is review from Alfonso still required here?

**Comment by Meulengracht** [comment]
I think it's okay to land

File: sysconfig/cloudinit.go
Comment (ernestl) [inline sysconfig/cloudinit.go:749]: Nitpick, "assume not disabled" does not seem accurate, we do not assume this,  but rely on "cloud-init" file based checks to determine CloudInitState?

**Review by ernestl — APPROVED** [review]

**Pull request closed without merging**
